main.go

```diff
@@ -1,343 +1,103 @@
 
-func main() {
-	if prettyPrint {
-		zlint.PrettyPrintZLint()
-		return
-	}
-	var err error
-	var theJSON []byte
-	if multi { //multiple cert file, do chunk reading method or multi-threaded method
-		if threaded {
-			threadMode()
-		} else {
-			err = multiMode() //write happens internally
-			if err != nil {
-				fmt.Println(err)
-			}
+func ProcessCertificate(in <-chan string, wg *sync.WaitGroup) {
+	log.Info("Processing certificates...")
+	defer wg.Done()
+	for raw := range in {
+		der, err := base64.StdEncoding.DecodeString(raw)
+		if err != nil {
+			//Handle error
 		}
-		return //no further work
-	} else { //single cert, read whole file
-		theJSON, err = singleMode()
-	}
-
-	if err != nil {
-		fmt.Println("Something went wrong....")
-		fmt.Println(err)
-	}
-
-	if outPath != "" && outPath != "-" {
-		ioutil.WriteFile(outPath, theJSON, 0644)
-	} else {
-		fmt.Println(string(theJSON))
-	}
-
-}
-
-func multiMode() (err error) {
-	//reader variables
-	var fileReader io.Reader //base file reader, to be buffered
-	fileReader, err = os.Open(inPath)
-	if err != nil { //file open/read error
-		return err
-	}
-	var buffReader *bufio.Reader = bufio.NewReader(fileReader) //buffered file reader from base reader
-	//writer
-	var fileWriter *os.File
-	if outPath != "" && outPath != "-" { //only open file if path specified
-		fileWriter, err = os.OpenFile(outPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0666) //os.File in write mode
-		if err != nil {                                                                  //file open/write error
-			return err
-		}
-		defer fileWriter.Close()
-	}
-	//main computation
-	var done bool = false
-	for !done { //loops until read error, consider reworking this
-		certs := make([]string, 0, CHUNKSIZE)              //input buffer
-		reports := make([]map[string]string, 0, CHUNKSIZE) //output buffer
-		var readIn, lintOut int = 0, 0                     //reading and processing counters
-		for ; readIn < CHUNKSIZE && !done; readIn++ {      //read in CHUNKSIZE # certs
-			lineIn, err := buffReader.ReadString('\n')
-			if err != nil { //read error, stop reading and process anything that was read
-				done = true
-				if err != io.EOF {
-					fmt.Println(err)
-					break //if not eof, assume incomplete cert read
-				}
-			}
-			if len(lineIn) > 0 { //dont add empty line at the end
-				certs = append(certs, lineIn) //add cert to process queue
-			}
-		}
-
-		//process read certs
-
-		for x := 0; x < len(certs); x++ {
-			reportOut, err := zlint.Lint64(certs[x]) //lint the cert
+		parsed, err := x509.ParseCertificate(der)
+		if err != nil { //could not parse
+			//log.Info("Could not parse certificate with error:", err)
+		} else { //parsed
+			zlintResult := zlint.ZLintResultTestHandler(parsed)
+			jsonResult, err := json.Marshal(zlintResult.ZLint)
 			if err != nil {
-				fmt.Println(err)
-			} else {
-				reports = append(reports, reportOut) //move report to out buffer
-				lintOut++
+				log.Fatal("Could not parse JSON.")
```

Comment:
@dadrian Did it crash though? Since this is particularly for the `json.Marshal` of the ZLint output. What is the behavior we want when a certificate can't be parsed?